pkg/storegateway/indexheader/reader_pool.go

```diff
@@ -51,21 +55,83 @@ type ReaderPool struct {
 	lazyReaders   map[*LazyBinaryReader]struct{}
 }
 
-// NewReaderPool makes a new ReaderPool and starts a background task for unloading idle Readers if enabled.
-func NewReaderPool(logger log.Logger, lazyReaderEnabled bool, lazyReaderIdleTimeout time.Duration, metrics *ReaderPoolMetrics) *ReaderPool {
+type HeadersLazyLoaded struct {
+	Path   string
+	UserID string
+}
+
+type HeadersLazyLoadedTrackerState struct {
+	LazyLoadedBlocks map[string]int64 `json:"lazy_loaded_blocks"`
+	UserID           string           `json:"user_id"`
+}
+
+func (p *ReaderPool) persist(state HeadersLazyLoadedTrackerState, finalPath string) error {
```

Comment:
Is there a reason why this is a method on `ReaderPool`? As far as I can tell, it doesn't use anything from `ReaderPool`. Perhaps it would fit better as a method on `HeadersLazyLoadedTrackerState`?

Reply:
That's true. I will do the small refactoring and see.